packages/eas-cli/src/commands/build/cancel.ts

```diff
@@ -0,0 +1,129 @@
+  const data = await withErrorHandlingAsync(
+    graphqlClient
+      .mutation<{ build: { cancel: { id: string; status: BuildStatus } } }>(
+        gql`
+          mutation CancelBuildMutation($buildId: ID!) {
+            build(buildId: $buildId) {
+              cancel {
+                id
+                status
+              }
+            }
+          }
+        `,
+        { buildId }
+      )
+      .toPromise()
+  );
+  return data.build!.cancel;
+}
+
+const appPlatformMap = {
+  [AppPlatform.Android]: Platform.Android,
+  [AppPlatform.Ios]: Platform.iOS,
+};
+
+function formatUnfinishedBuild(
+  build: Pick<Build, 'id' | 'platform' | 'status' | 'createdAt'>
+): string {
+  const platform = platformEmojis[appPlatformMap[build.platform]];
+  const startTime = new Date(build.createdAt).toLocaleString();
+  const status = chalk.blue(build.status === BuildStatus.InQueue ? 'in-queue' : 'in-progress');
+  return `${platform} Started at: ${startTime}, Status: ${status}, Id: ${build.id}`;
+}
+
+async function selectBuildToCancelAsync(
+  projectId: string,
+  fullProjectName: string
+): Promise<string | null> {
+  const spinner = ora().start('Fetching the unfinished builds…');
+  let builds;
+  try {
+    const [inQueueBuilds, inProgressBuilds] = await Promise.all([
+      BuildQuery.allForAppAsync(projectId, { status: BuildStatus.InQueue }),
+      BuildQuery.allForAppAsync(projectId, { status: BuildStatus.InProgress }),
+    ]);
+    spinner.stop();
+    builds = [...inQueueBuilds, ...inProgressBuilds];
+  } catch (error) {
+    spinner.fail(
+      `Something went wrong and we couldn't fetch the builds for the project ${fullProjectName}.`
+    );
+    throw error;
+  }
+  if (builds.length === 0) {
+    log.warn(`There are no unfinished builds for the project ${fullProjectName}.`);
+    return null;
+  } else if (builds.length === 1) {
+    log('Found one build');
+    log(formatUnfinishedBuild(builds[0]));
+    await confirmAsync({
+      message: 'Do you want to cancel it?',
+    });
+    return builds[0].id;
+  } else {
+    const buildId = await selectAsync<string>(
+      'Which build do you want to cancel?',
+      builds.map(build => ({
+        title: formatUnfinishedBuild(build),
+        value: build.id,
+      }))
+    );
+    log(buildId);
+    return buildId;
+  }
+}
+
+export default class BuildCancel extends Command {
+  static description = 'cancel an unfinished build';
+
+  static args = [{ name: 'BUILD_ID' }];
+
+  async run() {
+    const { buildId: buildIdFromArg } = this.parse(BuildCancel).args;
+
+    const projectDir = (await findProjectRootAsync()) ?? process.cwd();
+    const projectId = await getProjectIdAsync(projectDir);
+    const fullProjectName = await getProjectFullNameAsync(projectDir);
+
+    const buildId = buildIdFromArg || (await selectBuildToCancelAsync(projectId, fullProjectName));
+    if (!buildId) {
+      return;
+    }
+
```

Comment:
What if the user passed the build id as an argument but it's not valid / it doesn't exist in the database?

Reply:
sql error was printed in that case, I added fetch above and better error message